coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java

```diff
@@ -121,35 +129,39 @@ public SimpleClusterManager(CoordinatorConf conf, Configuration hadoopConf) thro
   void nodesCheck() {
     try {
       long timestamp = System.currentTimeMillis();
-      Set<String> deleteIds = Sets.newHashSet();
-      Set<String> unhealthyNode = Sets.newHashSet();
       for (ServerNode sn : servers.values()) {
         if (timestamp - sn.getTimestamp() > heartbeatTimeout) {
           LOG.warn("Heartbeat timeout detect, " + sn + " will be removed from node list.");
-          deleteIds.add(sn.getId());
+          sn.setStatus(ServerStatus.LOST);
+          lostNodes.add(sn);
+          unhealthyNodes.remove(sn);
         } else if (!sn.isHealthy()) {
           LOG.warn("Found server {} was unhealthy, will not assign it.", sn);
-          unhealthyNode.add(sn.getId());
+          unhealthyNodes.add(sn);
```

Comment:
Should we remove the node from `lostNodes` after this line?

Reply:
yes,This is actually going to be removed